### PR TITLE
feat: search v2 type filters

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/8916

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/omnibar/api/search.ts [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
     projectUuid: string;
     query: string;
-}) =>
-    lightdashApi<SearchResults>({
-        url: `/projects/${projectUuid}/search/${encodeURIComponent(query)}`,
+    filters?: SearchFilters;
+}) => {
+    const searchParams = filters
+        ? '?' + new URLSearchParams(filters).toString()
+        : '';
```
Comment (almeidabbm) [inline packages/frontend/src/features/omnibar/api/search.ts:15]:
```suggestion
    const searchParamsString = new URLSearchParams(filters).toString();
    const searchParams = searchParamsString && `?${searchParamsString}`;
```

The current condition will break if `filters` is an empty object. If that ever happens, then searchparams will be `?`

After trying it out, `new URLSearchParams(filters).toString()` will always return `''` if either `filters` is empty object or undefined, so it is safe to call without check and do the check just for the `searchParams` string on the next line

**Review by almeidabbm — CHANGES_REQUESTED** [review]

**Comment by almeidabbm** [comment]
I believe this is a little bug:


https://github.com/lightdash/lightdash/assets/22939015/4da75c8b-a4b9-4523-b4d7-d31a69569d90

**Comment by almeidabbm** [comment]
> I believe this is a little bug:
> 
>  Screen.Recording.2024-02-23.at.09.34.25.mov

fixed here: https://github.com/lightdash/lightdash/pull/9076

File: packages/frontend/src/features/omnibar/api/search.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/features/omnibar/api/search.ts]: fixed

**Review by almeidabbm — APPROVED** [review]

**Pull request merged**